Aws batch hillshade

MDavidson17 opened the pull request:
batch process to create hillshades. 

not included:
- ci
- testing

MDavidson17 committed:
feat: hillshade batch scripts

MDavidson17 committed:
feat: assume give batch instance role bucket access

MDavidson17 committed:
fix: fix script

MDavidson17 committed:
fix: allow asc files

and increase job size to avoid too many requests exception

blacha commented on aws-batch-processing-hillshade/batch/Dockerfile:
```diff
@@ -0,0 +1,14 @@
+RUN rm /bin/sh && ln -s /bin/bash /bin/sh
+
+# install awscli
+RUN apt update && apt install awscli -y
```
combine apt installs into one install `apt install awscli jq -y`

its also a good idea to clean up the packages after install, I cant remember the exact command but its something like `&& apt-cache-clean`

blacha commented on aws-batch-processing-hillshade/batch/script.sh:
```diff
@@ -0,0 +1,32 @@
+#!/bin/bash
+cd ./app
+tmp_dir=$(mktemp -d -t hs-XXXXXXXXXX)
```
generally use long form arguments to make it easier to read `-d` vs `--directory`

blacha commented on aws-batch-processing-hillshade/src/infra/batch.ts:
```diff
@@ -0,0 +1,63 @@
+import cdk = require('@aws-cdk/core');
```
never use `require`

```typescript
import * as cdk from '@aws-cdk/core
```

blacha commented on aws-batch-processing-hillshade/src/infra/submit.ts:
```diff
@@ -0,0 +1,90 @@
+  let string = ''
+  for await (let file_name of fsa.list(ReadFromFolder)) {
+    // let re = /\ /gi;
+    // file_name = file_name.replace(re, '\\ ')
```
would be a good idea to simplify this down
```
const goodExtension = new Set('tif', 'tiff', 'asc')
cosnt ext = extname(file_name)

if (ext in file_extensions) {
   file_name.replace(ext, 'tiff')
}
```

blacha commented on aws-batch-processing-hillshade/src/infra/submit.ts:
```diff
@@ -0,0 +1,90 @@
+      fileList = fileList.concat(string)
+      count++
+    }
+    if (count == 25) {
```
```
const RecordsPerJob = 25

if (fileList.length >= RecordsPerJob)
```

blacha commented on aws-batch-processing-hillshade/src/infra/submit.ts:
```diff
@@ -0,0 +1,90 @@
+    }
+  }
+  if (count > 0) {
+    submit(correlationId, "final-upload-" + upload_count, fileList)
```
await

blacha commented on aws-batch-processing-hillshade/src/infra/submit.ts:
```diff
@@ -0,0 +1,90 @@
+  let base = ''
+  let string = ''
+  for await (let file_name of fsa.list(ReadFromFolder)) {
+    // let re = /\ /gi;
```
The escaping is a problem in bash, I would leave these names alone

MDavidson17 committed:
fix: enhancements 

added await submit
fix handling of file list 
convert to python script
tidy up and delete temp_dir
added logging

MDavidson17 committed:
fix: remove shell script

replaced by python script

paulfouquet commented on aws-batch-processing-hillshade/batch/script.py:
```diff
@@ -0,0 +1,44 @@
+    os.environ["AWS_SECRET_ACCESS_KEY"]=credentials['SecretAccessKey']
+    os.environ["AWS_SESSION_TOKEN"]=credentials['SessionToken']
+
+temp_dir = tempfile.mkdtemp()
```
The indentation is weird here?

paulfouquet reviewed (changes requested):
Does the script work with the bad indentation?
